### DISCUSSION TITLE
Renovate doesn't seem to pass platform config to globalExtends

### DISCUSSION THREAD

**Discussion by BenWhiting** [renovatebot/renovate#21498] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.19

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

 Bitbucket v7.17.15

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I might be using and understanding globalExtends configuration wrong, but what I am experiencing is the following.

**Expected**
When I provide a platform and endpoint configuration definition at the root self hosted configuration level, those settings would be passed to imported globalExtends presets.

**Configuration**

Self-hosted configuration passed to renovate at runtime:
```
module.exports = {
  "platform": "bitbucket-server",
  "endpoint": "{company_internal_url}",
  "globalExtends": [
    "local>{project}/{repo}//{path}/onboarding"
   ]
 }
```

more self-hosted configurations I want to put in a preset to share with others:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "onboarding": true,
  "requireConfig": "optional",
  "onboardingBranch": "renovate/configure",
  "onboardingConfigFileName": ".renovaterc",
  "onboardingPrTitle": "Onboarding Renovate with local configuration",
  "onboardingNoDeps": true,
  "detectHostRulesFromEnv": true
}
```

Notes: 
- preset is hosted on the same platform as the target repository
- preset is inside of a private bitbucket project/repo that can successfully be pulled from during the processing of the extends settings

**Actual**

It seems that when I use globalExtends, the root configuration isn't processed until later causing the attached error.

**Error**
```
Preset fetch error
19:37:43         "preset": "local>{project}/{repo}//{path}/onboarding",
19:37:43         "err": {
19:37:43           "message": "Missing platform config for local preset.",
19:37:43           "stack": "Error: Missing platform config for local preset.\n    at Object.getPreset (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/config/presets/local/index.ts:37:11)\n    at getPreset (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/config/presets/index.ts:225:54)\n    at fetchPreset (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/config/presets/index.ts:359:18)\n    at resolveConfigPresets (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/config/presets/index.ts:291:37)\n    at resolveGlobalExtends (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/workers/global/index.ts:102:54)\n    at /opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/workers/global/index.ts:127:17\n    at start (/opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/workers/global/index.ts:120:5)\n    at /opt/buildpack/tools/renovate/35.19.0/node_modules/renovate/lib/renovate.ts:18:22"
19:37:43         }
```

### What you tried so far.

I haven't had anything work so far and moved the settings I wanted in a preset, into the root self-hosted configuration as a work around.

1. Moved platform and endpoint definitions to environment variables - maybe there was a processing order issue? same debug output in description.
2. Added platform and endpoint definitions to the preset I wanted to import through  globalExtends - same debug output in description.
3. Added platform and endpoint definitions at the top of the root self-hosted configuration.
4. Moved the globalExtends entry and extends entry - All self-hosted settings were ignored and renovate used the default hardcoded settings.

After some research this issue seems very similar to the follow issue: https://github.com/renovatebot/renovate/issues/14828 & might have to do with this logged discussion: https://github.com/renovatebot/renovate/discussions/18894 ?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
global extends presets can't be private

**Reply by BenWhiting** [reply]
Cool. Thank you for that!

Did I just miss that in the documents? I don't remember seeing that

**Reply by viceice** [reply]
no, it's not yet documented.

@HonkingGoose can you open a PR to add a note for that limitation?

**Reply by HonkingGoose** [reply]
See:

- #21597
